File: java/sdk/sdk-pay/src/main/java/com/novax/sdk/pay/receipt/ReceiptConfirmRequest.java
package com.novax.sdk.pay.receipt;

import com.novax.sdk.core.request.AbstractApiRequest;
import com.novax.sdk.core.request.HttpMethod;
import com.novax.sdk.core.request.TypeRef;
import lombok.Builder;
import lombok.Getter;

import java.util.LinkedHashMap;
import java.util.Map;

/** POST /pay/v3/receipt/order/confirm — confirm a receipt order. */
@Getter
@Builder
public class ReceiptConfirmRequest extends AbstractApiRequest<Object> {

    private final String protocol;
    private final String currency;
    private final String smartContractAddress;
    private final String companyUserId;
    private final String receiptOrderId;

    @Override public HttpMethod method() { return HttpMethod.POST; }
    @Override public String path() { return "/pay/v3/receipt/order/confirm"; }

    @Override public Object body() {
        Map<String, Object> b = new LinkedHashMap<>();
        b.put("protocol", protocol);
        b.put("currency", currency);
        b.put("smartContractAddress", smartContractAddress);
        b.put("companyUserId", companyUserId);
        b.put("receiptOrderId", receiptOrderId);
        return b;
    }

    @Override public TypeRef<Object> responseType() {
        return TypeRef.of(Object.class);
    }
}

File: python/novax_sdk/pay/receipt/receipt_confirm_request.py
from typing import Any, Optional

from novax_sdk.request.api_request import ApiRequest, HttpMethod


class ReceiptConfirmRequest(ApiRequest):
    """POST /pay/v3/receipt/order/confirm"""

    def __init__(self, protocol: Optional[str] = None, currency: Optional[str] = None,
                 smart_contract_address: Optional[str] = None,
                 company_user_id: Optional[str] = None,
                 receipt_order_id: Optional[str] = None) -> None:
        self._protocol = protocol
        self._currency = currency
        self._smart_contract_address = smart_contract_address
        self._company_user_id = company_user_id
        self._receipt_order_id = receipt_order_id

    @property
    def method(self) -> HttpMethod:
        return HttpMethod.POST

    @property
    def path(self) -> str:
        return "/pay/v3/receipt/order/confirm"

    def body(self) -> dict[str, Any]:
        return {
            "protocol": self._protocol,
            "currency": self._currency,
            "smartContractAddress": self._smart_contract_address,
            "companyUserId": self._company_user_id,
            "receiptOrderId": self._receipt_order_id,
        }

    @property
    def response_type(self):
        return None
